Assets/Scripts/App.cs

```diff
@@ -1895,57 +1895,71 @@ private void MoveUserDataFromOldLocation()
 
             if (Directory.Exists(m_UserPath))
             {
-                return;
+                if(File.Exists(Path.Combine(m_UserPath, "MovedFiles.txt")))
+                {
+                    // Files have previously been moved.
+                    return;
+                }
             }
 
             try
             {
                 // Moving does not work across different mount points, have to copy and delete.
-                CopyDirectory(m_OldUserPath, m_UserPath, true);
+                var movedFiles = MoveDirectory(m_OldUserPath, m_UserPath, true);
+
+                // If successful, we can delete the old directory.
                 Directory.Delete(m_OldUserPath, true);
+
                 // Recreate the old directory and put a message in there so a user used to looking in the old
                 // location can find out where to get their files.
                 Directory.CreateDirectory(m_OldUserPath);
                 string moveMessageFilename = Path.Combine(m_OldUserPath, kFileMoveFilename);
                 File.WriteAllText(moveMessageFilename, kFileMoveContents);
+
+                // Signify this operation is comlpete by storing the manifest in the new directory.
+                File.WriteAllText(Path.Combine(m_UserPath, "MovedFiles.txt"), String.Join('\n', movedFiles.ToArray()));
+
             }
             catch (Exception ex)
             {
                 Debug.LogException(ex);
             }
         }
 
-        static void CopyDirectory(string sourceDir, string destinationDir, bool recursive)
+        static List<string> MoveDirectory(string sourceDir, string destinationDir, bool recursive)
         {
-            // Get information about the source directory
             var dir = new DirectoryInfo(sourceDir);
 
-            // Check if the source directory exists
+            List<string> movedFiles = new List<string>();
+
             if (!dir.Exists)
                 throw new DirectoryNotFoundException($"Source directory not found: {dir.FullName}");
 
-            // Cache directories before we start copying
             DirectoryInfo[] dirs = dir.GetDirectories();
 
-            // Create the destination directory
             Directory.CreateDirectory(destinationDir);
 
-            // Get the files in the source directory and copy to the destination directory
             foreach (FileInfo file in dir.GetFiles())
             {
                 string targetFilePath = Path.Combine(destinationDir, file.Name);
-                file.CopyTo(targetFilePath);
+                if(!File.Exists(targetFilePath))
```

Comment:
What if the file exists in both places? I can see two scenarios in which this will happen:

1. The copy was interrupted, and might be a partial copy
2. They loaded an old version and saved it in the Open Brush directory.

If it's the former, we should attempt a copy with overwrite.
If it's the latter, we should attempt a copy with a rename.

Since we have no way to differentiate, can we copy but to a new name? E.g., from `foo.tilt`. to `foo.1.tilt`? (similar to wget's suffixing?)

See also the comment about the MovedFiles.txt in general above.

Reply:
Good idea! tempted to break out a checksum too...

Reply:
If the checksum is the same, you can just delete. But if it's not, we'd still want to differentiate between identical files that were only partially copied and the downgrade case (or even someone who has multiple versions installed).